### PR TITLE
golden middle between che-operator dependencies and mature enough controller-runtime version

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
golden middle between che-operator dependencies and mature enough controller-runtime version
CQs TODO after that 
- [x] github.com/alexflint/go-arg	v1.3.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22540
- [x] github.com/alexflint/go-scalar	v1.0.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22539
- [x] github.com/beorn7/perks	v0.0.0-20180321164747-3a771d992973	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18422
- [x] github.com/davecgh/go-spew	v1.1.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18423
- [x] github.com/go-logr/logr	v0.1.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19229
- [ ] github.com/go-logr/zapr	v0.1.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22560
- [x] github.com/golang/protobuf	v1.3.2	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20959
- [x] github.com/imdario/mergo	v0.3.6	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18434
- [x] github.com/json-iterator/go	v1.1.5	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16578
- [x] github.com/matttproud/golang_protobuf_extensions	v1.0.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18436
- [x] github.com/operator-framework/operator-sdk	v0.5.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19238
- [x] github.com/pkg/errors	v0.8.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19967
- [x] github.com/prometheus/client_golang	v0.9.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19243
- [x] github.com/spf13/pflag	v1.0.2	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17578
- [x] go.uber.org/atomic	v1.3.2	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19250
- [x] go.uber.org/multierr	v1.1.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19251
- [x] go.uber.org/zap	v1.9.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19252
- [x] gomodules.xyz/jsonpatch/v2	v2.0.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22561
- [x] gopkg.in/fsnotify.v1	v1.4.7	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20084
- [x] gopkg.in/inf.v0	v0.9.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22565
- [x] gopkg.in/yaml.v2	v2.2.2	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16568
- [x] k8s.io/klog	v0.3.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20464
- [ ] k8s.io/utils	v0.0.0-20190506122338-8fab8cb257d5	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22564
- [x] sigs.k8s.io/controller-runtime	v0.2.1	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22563
- [x] sigs.k8s.io/yaml	v1.1.0	https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22562
Logging taken from  https://github.com/che-incubator/configbump/pull/9 is working

Знімок екрана 2020-08-27 о 14 51 39

 
### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17568

### CODE REVIEW

File: pkg/configmaps/lib.go [hunk 1]
```diff
@@ -72,16 +71,16 @@ func New(mgr manager.Manager, config ConfigMapReconcilerConfig) (controller.Cont
...
-	// the label has been removed
-	//bld.WithEventFilter(predicate.ResourceFilterPredicate{Selector: r.selector})
-	ctrl, err := bld.Build(r)
+	ctrl, err := controller.New("config-bump", mgr, controller.Options{Reconciler: r})
+	err = ctrl.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &corev1.ConfigMap{},
+	})
+
+	if err != nil {
```
Comment (metlos) [inline pkg/configmaps/lib.go:80]: Duplicated error handling?

**Review by metlos — COMMENTED** [review]
I promise to never use any new version of anything in Eclipse projects :smile: Thanks for sorting the mess..

**Comment by skabashnyuk** [comment]
tests not working.  Checking that ....

**Comment by skabashnyuk** [comment]
tests are working

**Review by sparkoo — APPROVED** [review]

**Review by metlos — APPROVED** [review]

**Pull request merged**